src/Giraffe/EndpointRouting.fs

```diff
@@ -0,0 +1,134 @@
+module Giraffe.EndpointRouting
+
+open System
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Http
+open Microsoft.AspNetCore.Routing
+open Microsoft.FSharp.Reflection
+open FSharp.Core
+open FSharp.Control.Tasks.V2.ContextInsensitive
+
+module private RouteTemplateBuilder =
+    let private getConstraint (i : int) (c : char) =
+        // Route Constraint Docs:
+        // https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-3.0#route-constraint-reference
+        (dict [
```

Comment:
Are we able to just convert this to a match statement instead? this is creating a new dictionary every time its called which is a bit wasteful and rather then lifting it to a static value, since all values are constant / known at compile time, match statements should always be used as match will be faster then dictionary 99% of the time thanks to jump table optimizations and branch locality